docs/dev/building/Compile.rst

```diff
@@ -0,0 +1,456 @@
+
+    git checkout master
+    git submodule update
+
+In general, a single DFHack clone is suitable for development - most Git
+operations such as switching branches can be done on an existing clone. If you
+find yourself cloning DFHack frequently as part of your development process, or
+getting stuck on anything else Git-related, feel free to reach out to us for
+assistance.
+
+.. admonition:: Offline builds
+
+  If you plan to build DFHack on a machine without an internet connection (or
+  with an unreliable connection), see `note-offline-builds` for additional
+  instructions.
+
+.. admonition:: Working with submodules
+
+  DFHack uses submodules extensively to manage its subprojects (including the
+  ``scripts`` folder and DF-structures in ``library/xml``). Failing to keep
+  submodules in sync when switching between branches can result in build errors
+  or scripts that don't work. In general, you should always update submodules
+  whenever you switch between branches in the main DFHack repo with
+  ``git submodule update``. (If you are working on bleeding-edge DFHack and
+  have checked out the master branch of some submodules, running ``git pull``
+  in those submodules is also an option.)
+
+  Rarely, we add or remove submodules. If there are any changes to the existence
+  of submodules when you switch between branches, you should run
+  ``git submodule update --init`` instead (adding ``--init`` to the above
+  command).
+
+  Some common errors that can arise when failing to update submodules include:
+
+  * ``fatal: <some path> does not exist`` when performing Git operations
+  * Build errors, particularly referring to structures in the ``df::`` namespace
+    or the ``library/include/df`` folder
+  * ``Not a known DF version`` when starting DF
+  * ``Run 'git submodule update --init'`` when running CMake
+
+  Submodules are a particularly confusing feature of Git. The
+  `Git Book <https://git-scm.com/book/en/v2/Git-Tools-Submodules>`_ has a
+  thorough explanation of them (as well as of many other aspects of Git) and
+  is a recommended resource if you run into any issues. Other DFHack developers
+  are also able to help with any submodule-related (or Git-related) issues
+  you may encounter.
+
+Dependencies
+------------
+
+If you haven't already checked, this would be a good point to ensure you have
+all the `dependencies <build-dependencies>`.
+
+
+Contributing to DFHack
+----------------------
+
+For details on contributing to DFHack, including pull requests, code
+format, and more, please see `contributing-code`.
+
+
+Build settings
+==============
+
+This section describes build configuration options that apply to all platforms.
+If you don't have a working build environment set up yet, follow the instructions
+in the platform-specific sections below first, then come back here.
+
+Be sure to check out common `build options <build-options>`.
+
+Instructions
+============
+
+.. contents::
+  :local:
+  :depth: 1
+
```

Comment:
since this is the main focus of the page now, I think we can get rid of this header and move everything below it up a header level.

Reply:
I disagree, this makes for a good landing spot for anyone looking for just the instructions. The information would of course be visible from the top level, but it does add clutter. Have you taken a look at the html? The index links to such sections.

Reply:
I did most of the reviewing from the generated HTML. The current TOC on the Building page looks like this:
```
Compilation
How to get the code
Build settings
Instructions
```

I'm saying this would save readers a click if instead of "Instructions" we let them directly access the instructions for the platform they're interested in.

Reply:
You've convinced me actually. I had considered this originally, but decided against it because I wanted to reinforce that there are sections relevant to everyone. I thought all the submodules being a gotcha was a good reason to avoid adding platforms for users to instantly spot and click and end up not populating submodules. So that was my rational, if you agree with it I'll stick with it otherwise move them up a level

Reply:
yeah, I see what you're saying, and that's a good point. How about this:
- Keep `How to get the code` at the top
- Remove the `Dependencies` and `Build Settings` sections (which are both just stubs pointing to other pages)
- Move the platforms up a level (get rid of the `Instructions` header)
- At the start of each platform section, have a short sentence like:
```
Be sure you have `downloaded the code <How to get the code>`_, installed the `build-dependencies`, and configured the `build-options`.
```
Then even if they click straight to their platform, they won't miss common information.

Reply:
That would take us back to that idea of "including the same text in multiple locations"

Reply:
yeah, but this "repeated text" is just hyperlinks to other sections, not actual original content. I don't think that crosses the threshold for "duplication"